src/Game.js

```diff
@@ -0,0 +1,209 @@
+ */
+var Game = function(player1, player2) {
+  /**
+   * A reference to the player1 object instance
+   *
+   * @property player1
+   * @type {Object}
+   * @default undefined
+   */
+  this.player1 = player1 || {};
+
+  /**
+   * A reference to the player2 object instance
+   *
+   * @property player2
+   * @type {Object}
+   * @default undefined
+   */
+  this.player2 = player2 || {};
+
+  /**
+   * An array representing the tic tac toe game board
+   *
+   * @property gameBoard
+   * @type {Array}
+   * @default [0,0,0,0,0,0,0,0,0]
+   */
+  this.gameBoard = [0,0,0,0,0,0,0,0,0];
+
+  /**
+   * Tracks whether the game is won by either player
+   *
+   * @property gameIsWon
+   * @type {Boolean}
+   * @default true
+   */
+  this.gameIsWon = false;
+
+  /**
+   * Tracks whether the game is a draw
+   *
+   * @property gameIsDrawn
+   * @type {Boolean}
+   * @default true
+   */
+  this.gameIsDrawn = false;
+
+  /**
+   * A reference to the PubSub object instance
+   *
+   * @property pubSub
+   * @type {Object}
+   * @default undefined
+   */
+  this.pubSub = new PubSub() || {};
+
+  this.init();
+};
+
+/**
+ * Initializes the game engine
+ *
+ * @method init
+ * @returns Game
+ */
+Game.prototype.init = function() {
+  this.assignSides()
+      .runGame(this.player1);
+
+  return this;
+};
+
+/**
+ * Assigns tic tac toe sides to the players
+ *
+ * @method assignSides
+ * @returns Game
+ * @chainable
+ */
+Game.prototype.assignSides = function() {
+  this.player1.side = 1;
+  this.player2.side = 2;
+
+  return this;
+};
+
+/**
+ * Recursively runs getMove methods on the players until
+ * there is a winner
+ *
+ * @method runGame
+ * @param {Object} currentPlayer The current player of the game
+ */
+Game.prototype.runGame = function(currentPlayer) {
+  var move;
+  var nextPlayer;
+
+  if (currentPlayer === this.player1) {
+    nextPlayer = this.player2;
+  } else {
+    nextPlayer = this.player1;
+  }
+
+  if (!this.gameIsWon && !this.gameIsDrawn) {
+    move = currentPlayer.getMove(this.gameBoard, currentPlayer.side);
+    this.updateGameBoard(move, currentPlayer);
+    this.checkForWin();
+    this.checkForDraw();
+    this.runGame(nextPlayer);
+  } else if (this.gameIsWon){
+    this.publishWin(currentPlayer.side);
+  } else if (this.gameIsDrawn){
+    this.publishDraw();
+  }
+};
+
+/**
+ * Updates the game board with the current player move
+ *
+ * @method updateGameBoard
+ * @param {int} move The index of the game board array where the move is made
+ * @param {Object} currentPlayer The current player of the game
+ * @chainable
+ */
+Game.prototype.updateGameBoard = function(move, currentPlayer) {
+  this.gameBoard[move] = currentPlayer.side;
+
+  this.pubSub.publish('gameUpdate', {
+    gameBoard: this.gameBoard
+  });
+
+  return this;
+};
+
+/**
+ * Checks the game board to see if the current player's move wins the game
+ *
+ * @method checkForWin
+ * @chainable
+ */
+Game.prototype.checkForWin = function() {
+  switch(true) {
```

Comment:
I challenge you to make this an algorithm instead of brute forcing it. Imagine that we would expand this to any size board (4x4, 5x5, 100x100)